### PR TITLE
Hotshot/rc 0.5.60

### PR DESCRIPTION
**Pull request by jparr721**
Closes #<ISSUE_NUMBER>

### CODE REVIEW

File: sequencer/src/lib.rs [hunk 1]
```diff
@@ -762,6 +767,10 @@ pub mod testing {
             self.builder_port
         }
 
+        pub fn url(&self) -> Url {
+            self.url.clone()
+        }
```
Comment (jbearer) [inline sequencer/src/lib.rs:772]: This should probably be called `l1_url` as that's what it really represents. I'm not sure why it had such a generic name as `url` to begin with, as this is a struct representing an entire network and just `url` is really vague

File: sequencer/src/lib.rs [hunk 2]
```diff
@@ -589,23 +589,24 @@ pub mod testing {
...
         port: Option<u16>,
-    ) -> (Box<dyn BuilderTask<SeqTypes>>, Url) {
-        let builder_config = if let Some(port) = port {
-            SimpleBuilderConfig { port }
-        } else {
-            SimpleBuilderConfig::default()
-        };
+    ) -> Box<dyn BuilderTask<SeqTypes>> {
+        let port = port.unwrap_or_else(|| pick_unused_port().expect("No available ports"));
+        url.set_port(Some(port)).expect("Failed to set port");
```
Comment (jbearer) [inline sequencer/src/lib.rs:603]: Unclear why we're taking a URL just to change the port, vs just `localhost:{port}`

**Review by jbearer — CHANGES_REQUESTED** [review]

File: sequencer/src/lib.rs (hunk 2 above)
Comment (jparr721) [reply sequencer/src/lib.rs:603]: Good point.. honestly not sure why that was there.

File: sequencer/src/lib.rs [hunk 3]
```diff
@@ -596,12 +596,8 @@ pub mod testing {
     const STAKE_TABLE_CAPACITY_FOR_TEST: u64 = 10;
 
     pub async fn run_test_builder<const NUM_NODES: usize>(
-        mut url: Url,
-        port: Option<u16>,
+        url: Url,
```
Comment (jbearer) [inline sequencer/src/lib.rs:599]: I think I meant the opposite suggestion. We should pick an unused port (or have one passed in) and the url we use for the builder is just `localhost:{that port}`. Now it looks like we are using the same URL for the builder and L1 in most tests which isn't going to work

**Review by jbearer — APPROVED** [review]

**Pull request merged**